### PR TITLE
removed request library dependency by using _oauth.request method for making _oauth.get calls

### PR DESCRIPTION
**Pull request by Lakhan-Nad**
removed request library dependency by using _oauth.request method for overwriting _oauth.get method, this created a minor change in callback declaration. The callback passed to _oauth.request is required to be of callback(err, body, res) form but in **userProfile** method the format was callback(err, res, body) so just interchanged the parameters.

**changes:**
- changed the callback provided to _oauth.get from (err, res,body) to (err, body, res)
- added example/test.js for testing the changes
- overwritten _oauth.get decalration using _oauth.request

**relevant links:**
- [the actual implementation of _oauth.get method](https://github.com/ciaranj/node-oauth/blob/master/lib/oauth2.js#L219)
- [the callback for of _oauth.request](https://github.com/ciaranj/node-oauth/blob/master/lib/oauth2.js#L134) : in this callback response maps to res and result maps to body.
- [the callback implemntation in passport-google-oauth20 lib](https://github.com/jaredhanson/passport-google-oauth2/blob/master/lib/strategy.js#L84)

### CODE REVIEW

File: examples/test.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+const passport = require("passport");
+const OutlookStrategy = require("../lib/strategy");
+
+passport.use(
+  new OutlookStrategy(
+    {
+      callbackURL: "http://localhost:3000/auth/outlook/redirect",
+      passReqToCallback: true,
+      clientID: "3d1dc301-4f26-4b9c-a641-640d237c5ebe",
+      clientSecret: "6o9sdc.u3eI-YQ_BGa4zeJ~G9_0mgbr2Tc",
```
Comment (clocked0ne) [inline examples/test.js]: If these are personal credentials and not provided by Microsoft you should redact them right away and set this using environment variables, perhaps adding a comment as such to the test.

**Comment by clocked0ne** [comment]
Thanks for contributing! This looks good aside from my comment above.

File: examples/test.js (hunk 1 above)
Comment (Lakhan-Nad) [reply examples/test.js]: Yes thanks! I will change it soon.

**Comment by Lakhan-Nad** [comment]
> Thanks for contributing! This looks good aside from my comment above.

Thank you!

**Comment by Lakhan-Nad** [comment]
Hello @clocked0ne. Is there any changes in the PR?

**Comment by clocked0ne** [comment]
> 
> 
> Hello @clocked0ne. Is there any changes in the PR?

No, but currently builds are failing, I want to make sure it is up to date and I can run tests this weekend before merging and releasing publicly.

**Comment by Lakhan-Nad** [comment]
> > Hello @clocked0ne. Is there any changes in the PR?
> 
> No, but currently builds are failing, I want to make sure it is up to date and I can run tests this weekend before merging and releasing publicly.

Okay, thanks for reply.

**Comment by clocked0ne** [comment]
@Lakhan-Nad Apologies for the delay, I have updated the PR with fixed tests and maintenance so the build passes, I will merge and publish this as a major update shortly (as the minimum version of NodeJS required has now changed to v10).

**Comment by Lakhan-Nad** [comment]
No issues Sir. Thanks for mention as contributor.

**Pull request merged**